src/app/pages/single-course-page/CoursePage.tsx

```diff
@@ -0,0 +1,174 @@
+import { AxiosError } from "axios";
+import Hls from "hls.js";
+import { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import {
+  coursePageSlice,
+  coursesListPageSlice,
+  useAppSelector,
+} from "../../redux";
+import { Client, IGetCourse } from "../../services";
+import { authErrorResponse } from "../../services/controllers/constants";
+import { Title, Wrapper, Description, BackButton } from "./styles";
+import { LessonCardComponent } from "../../components/lesson-card";
+import { CircularProgress } from "@mui/material";
+import { unlockedStatus } from "../../components/lesson-card/constants";
+
+export const CoursePage: React.FC = () => {
+  const { token, courseId } = useAppSelector(
+    (state) => state.persistedReducer.coursesListPage
+  );
+  const { coursesProgress } = useAppSelector(
+    (state) => state.persistedReducer.coursePage
+  );
+  const { setToken } = coursesListPageSlice.actions;
+  const { setCourseProgress, setLessonProgress } = coursePageSlice.actions;
+  const dispatch = useDispatch();
+
+  const client = new Client();
+  const navigate = useNavigate();
+  const [courseData, setCourseData] = useState({} as IGetCourse);
+
+  useEffect(() => {
+    async function getData() {
+      async function getToken() {
+        try {
+          const responseData = (await client.auth.getToken()).data;
+          dispatch(setToken(responseData.token));
+          return responseData.token;
+        } catch (error) {
+          console.log(error);
+          return;
+        }
+      }
+
+      async function getCourse(token: string) {
+        try {
+          const responseData = (await client.courses.getCourse(courseId, token))
+            .data;
+          const lessons = responseData.lessons.sort(
+            (a, b) => a.order - b.order
+          );
+          responseData.lessons = lessons.slice();
+          console.log(responseData);
```

Comment:
It recommended to avoid using of **`console.log()`** in your production code to improve performance, security, code cleanliness, and maintenance.

Reply:
Thank you! I agree, but thought it is acceptable for test case project.